src/main/java/com/idrsolutions/microservice/BaseServlet.java

```diff
@@ -214,6 +215,13 @@ protected void doPost(final HttpServletRequest request, final HttpServletRespons
             return;
         }
 
+        final String[] settings = request.getParameterMap().get("settings");
+        final SettingsValidator validator = validateSettings(settings[0]);
```

Comment:
@KieranFrance - From testing locally, if settings is not provided as a parameter for a request, this code causes a null pointer exception (trying to access the index of 0 on null). 

Since the validateSettings method is prepared to receive null values, I recommend changing to the following:

```suggestion
        final String[] settings = request.getParameterMap().get("settings");
        final SettingsValidator validator = settings != null ? validateSettings(settings[0]) : validateSettings(null);
```

Reply:
Changes made in following commit.